OZON.ru.js

```diff
@@ -1,396 +1,150 @@
-	var itemTypeDOMNode =
-	doc.evaluate(itemTypeXPath, doc, null,
-	XPathResult.ANY_TYPE, null).iterateNext();
-	var itemType;
-	if (itemTypeDOMNode) itemType = itemTypeDOMNode.textContent;
-	else return "";
-	if (itemType == "Книги") {
-		var nameXPath = '//div[@class="techDescription"]/div/div[2]/span';
-		var valueXPath = '//div[@class="techDescription"]/div/div[3]/span';
-		var nameXPathRes = doc.evaluate(
-		nameXPath, doc, null, XPathResult.ANY_TYPE, null);
-		var valueXPathRes = doc.evaluate(
-		valueXPath, doc, null, XPathResult.ANY_TYPE, null);
-		if (!IsAudioBook(nameXPathRes, valueXPathRes)) return "book";
-	} else {
-		itemTypeXPath = '//ul[@class="navLine"]/li[2]/a';
-		itemTypeDOMNode =
-		doc.evaluate(itemTypeXPath, doc, null,
-		XPathResult.ANY_TYPE, null).iterateNext();
-		if (itemTypeDOMNode) itemType = itemTypeDOMNode.textContent;
-		else return "";
-		if (itemType == "Цифровые книги") return "book";
-	}
-}
+	var hasIsbn = ZU.xpathText(doc, '//*[@itemprop="isbn"]') !== null;
+	var hasContent = ZU.xpath(doc, '//div[@class="bContentBlock"]').length > 0;
 
-function IsAudioBook(nameXPathRes, valueXPathRes) {
-	var nameDOMNode;
-	var valueDOMNode;
-	while( (nameDOMNode = nameXPathRes.iterateNext()) &&
-	(valueDOMNode = valueXPathRes.iterateNext()) ) {
-		var nameText = CleanText(nameDOMNode.textContent);
-		var valueText = CleanText(valueDOMNode.textContent);
-		if (nameText == "Формат аудиокниги") return true;
-	}
-	return false;
+	if (hasContent && hasIsbn) return 'book';
 }
 
-function MergeObjects(obj1, obj2) {
-	var result = new Object();
-	for (var attrName in obj1) {
-		result[attrName] = obj1[attrName];
-	}
-	for (var attrName in obj2) {
-		result[attrName] = obj2[attrName];
-	}
-	return result;
-}
+function doWeb(doc, url) {
+	var book = new Zotero.Item('book');
 
-function CleanText(text) {
-	var result = text.replace(/^\s*|\s*$/g, '');
-	result = result.replace(/\s+/g, ' ');
-	return result;
-}
+	var content = ZU.xpath(doc, '//div[@class="bContentBlock"]')[0];
 
-function CleanCreator(creatorStr) {
-	return CleanText(creatorStr);
-}
+	// basic info
+	book.title = ZU.xpathText(content, './h1[@itemprop="name"]');
+
+	book.publisher = ZU.xpathText(content, './/*[@itemprop="publisher"]/a');
+	book.numPages = ZU.xpathText(content, './/*[@itemprop="numberOfPages"]');
 
-function GetCreators(creatorsStr, creatorsCategory) {
-	var result = new Array();
-	var creator;
-	if (creatorsStr.match(",")) {
-		// http://www.ozon.ru/context/detail/id/7341440/
-		var authors = creatorsStr.split(",");
-		for (var i in authors) {
-			creator = CleanCreator(authors[i]);
-			creator = Zotero.Utilities.cleanAuthor(creator, creatorsCategory);
-			result.push(creator);
+
+	var isbnField = ZU.xpathText(content, './/*[@itemprop="isbn"]');
+	var isbnRe = /[\d\-]{9,}/g;
+	var isbn;
+	book.ISBN = [];
+	if (isbnField !== null) {
+		while ((isbn = isbnRe.exec(isbnField)) !== null) {
+			book.ISBN.push(ZU.cleanISBN(isbn[0]));
```

Comment:
`book.ISBN` needs to be a string. Convert the array to string afterwards, or use something like `book.ISBN += isbn[0] + ' ';` (again, you don't need to clean anything up, since Zotero will do it later anyway)